src/User/Controller/RateLimitToUserApiMiddleware.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\User\Controller;
+
+use Nyholm\Psr7\Factory\Psr17Factory;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use Yiisoft\Yii\RateLimiter\Counter;
+use Yiisoft\Yii\RateLimiter\LimitRequestsMiddleware;
+use Yiisoft\Yii\RateLimiter\Policy\LimitPerIp;
+use Yiisoft\Yii\RateLimiter\Storage\StorageInterface;
+
+class RateLimitToUserApiMiddleware implements MiddlewareInterface
```

Comment:
What is the benefit if it already exists?

Reply:
It is compatible with desire for coherence and modeling Consistency. This is the only place where there is rate limiting logic for the user api.

Reply:
This is not an ideal implementation and just demo code for subsequent implementations.